src/fmu/dataio/datastructure/_internal/internal.py

```diff
@@ -135,3 +140,29 @@ def _future_warning_property_seismic(self) -> AllowedContent:
         if isinstance(self.seismic, str):
             seismic_warn()
         return self
+
+
+class JsonSchemaMetadata(BaseModel, populate_by_name=True):
```

Comment:
Unsure with regards to how to structure the module/class hierarchy at this point. Hoping that the path will clearly revel itself as we add more of them.